src/build.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+import path from 'path';
+import {createWriteStream} from 'fs';
+import streamToPromise from 'stream-to-promise';
+
+import * as fs from './util/promised-fs';
+import {zipDir} from './util/zip-dir';
+
+
+export default function build({sourceDir, buildDir}: Object) {
```

Comment:
> To be clear, I think we should eventually add integration tests but only a very few -- i.e. to act more as "smoke tests" rather than tests that address the integration of components.

Yeah that's what I was thinking i.e. not running integration tests at all as part of the normal test suite. E.g. you could run unit tests day to day but only run the integration tests against PRs or merges.

> It will definitely require additional syntax because some types can't be inferred (it's a limitation of JavaScript). I did some searching and I see that comment-based syntax is supported but not really documented: http://flowtype.org/blog/2015/02/20/Flow-Comments.html At first it seemed like a good idea but adding a transpilation step is not a problem for us. I fear that relying on comments will just make the code harder to read.

The biggest issue I have is that the flow type annotations are not easy to parse to the point of making reading and reviewing annotated code much harder, plus it sort of looks like it could be some kind of es-next syntax which is worse imho. All in all that's going to confuse anyone looking at this code until they realise what it is. 

I wonder if a type checking lib might be easier to grok? - e.g: https://www.npmjs.com/package/type-check to just pick one example out of thin air. Although it's a bit crude it's seems much more obvious.

It's up to you but I'm beginning to feel that compiled non-standard extensions to JS are not a good idea - this is on the way to being typescript. Don't get me wrong; I see the problem - I'm just not sure this is a good solution.